File: tool4_junit/codeFiles/junit/src/junit/q2_string.java
package junit;

public class q2_string {
	boolean ispalindrome(String str1) {
		int len=str1.length()-1;
		int i=0;
		while(i<=len) {
			if(str1.charAt(i)!=str1.charAt(len)) {
				return false;
			}
			i++;
			len--;
		}
		return true;
	}
	
	boolean isSimilar(String str1,String str2) {
		int len1=str1.length();
		int len2=str2.length();
		if(len1!=len2) {
			return false;
		}
		for(int i=0;i<len1;i++) {
			if(str1.charAt(i)!=str2.charAt(i)) {
				return false;
			}
		}
		
		return true;
	}
}

File: tool4_junit/codeFiles/python/Q2_String.py
import unittest

class String:
    def is_palindrome(self, str1):
        len_str1 = len(str1) - 1
        i = 0
        while i <= len_str1:
            if str1[i] != str1[len_str1]:
                return False
            i += 1
            len_str1 -= 1
        return True

    def is_similar(self, str1, str2):
        len_str1 = len(str1)
        len_str2 = len(str2)
        if len_str1 != len_str2:
            return False
        for i in range(len_str1):
            if str1[i] != str2[i]:
                return False
        return True

class TestStringMethods(unittest.TestCase):
    def test_is_palindrome(self):
        s = String()
        self.assertTrue(s.is_palindrome("aba"))

    def test_is_similar(self):
        s = String()
        self.assertTrue(s.is_similar("hello", "hello"))

if __name__ == '__main__':
    unittest.main()
